libnopegl/src/utils.h

```diff
@@ -60,6 +60,16 @@
 #define NGLI_MAX(a, b) ((a) > (b) ? (a) : (b))
 #define NGLI_CLAMP(x, min, max) NGLI_MAX(NGLI_MIN(x, max), min)
 
+#define NGLI_ARRAY_MEMDUP(dst, src, name) do {                          \
+    size_t nb = (src).nb_##name;                                        \
+    if (nb > 0) {                                                       \
+        (dst).name = ngli_memdup((src).name, nb * sizeof(*(dst).name)); \
+        if (!(dst).name)                                                \
+            return NGL_ERROR_MEMORY;                                    \
+        (dst).nb_##name = nb;                                           \
```

Comment:
Assigning the count should probably be done outside the `if` for safety reasons (and maybe assign a `NULL` as well for the pointer)

Reply:
I assigned unconditionally the ptr to NULL and nb to 0 at the beginning.